lib/zlib.js

```diff
@@ -574,6 +574,16 @@ Zlib.prototype._processChunk = function(chunk, flushFlag, cb) {
   req.callback = callback;
 
   function callback(availInAfter, availOutAfter) {
+    // When the callback is used in an async write, the callback's
+    // context is the `req` object that was created. The req object
+    // is === this._handle, and that's why it's important to null
+    // out the values after they are done being used. `this._handle`
+    // can stay in memory longer than the callback and buffer are needed.
+    if (this) {
```

Comment:
@mdlavin `newReq === req` because `_handle.write()` just returns `_handle` (I know it’s confusing)… but yes you can leave it like this then.